test/pay/payments_test.exs

```diff
@@ -318,36 +318,33 @@ defmodule Pay.PaymentsTest do
       assert payment.gateway_transaction_id == "some gateway_transaction_id"
       assert payment.reference == "some reference"
       assert payment.return_url == "some return_url"
-      assert payment.status == "some status"
+      assert payment.status == "created"
       assert payment.wallet == "some wallet"
     end
 
     test "create_payment/1 with invalid data returns error changeset" do
       assert {:error, %Ecto.Changeset{}} = Payments.create_payment(@invalid_attrs)
     end
 
-    test "update_payment/2 with valid data updates the payment" do
+    test "update_payment/3 changes status according to transition" do
       {_, payment} = payment_fixture()
-      assert {:ok, %Payment{} = payment} = Payments.update_payment(payment, @update_attrs)
-      assert payment.amount == 43
-      assert payment.auth_3ds_details == %{}
-      assert payment.card_details == %{}
-      assert payment.delayed_capture == false
-      assert payment.description == "some updated description"
-      assert payment.email == "some updated email"
-      assert payment.external_id == "7488a646-e31f-11e4-aace-600308960662"
-      assert payment.external_metadata == %{}
-      assert payment.gateway_transaction_id == "some updated gateway_transaction_id"
-      assert payment.reference == "some updated reference"
-      assert payment.return_url == "some updated return_url"
-      assert payment.status == "some updated status"
-      assert payment.wallet == "some updated wallet"
+
+      assert {:ok, %Payment{} = updated_payment} =
+               Payments.update_payment(payment, "submit_payment", @update_attrs)
```

Comment:
@dagoof Should you get the available transitions of an "enum" not a string?

Reply:
yeah. I'm not really sure what the transitions will be yet. We've got hundreds from the uk that don't really seem to apply. I can enumify what we've got for now though